pkg/console/tlsconfig/tlsconfig_test.go

```diff
@@ -0,0 +1,228 @@
+
+		return tempFile.Name()
+	}
+
+	BeforeEach(func() {
+		tlsProfile := &v1alpha1.TlsSecurityProfile{
+			Type:         ocpconfigv1.TLSProfileIntermediateType,
+			Intermediate: &ocpconfigv1.IntermediateTLSProfile{},
+		}
+		tlsProfileJson, err := json.Marshal(tlsProfile)
+		Expect(err).ToNot(HaveOccurred())
+
+		certBytes, keyBytes, err := cert.GenerateSelfSignedCertKey(certHostName, nil, nil)
+		Expect(err).ToNot(HaveOccurred())
+
+		tlsConfigPath = createTempFile("vm-console-proxy-tls-config-*.json", tlsProfileJson)
+		certPath = createTempFile("vm-console-proxy-cert-*.crt", certBytes)
+		keyPath = createTempFile("vm-console-proxy-key-*.key", keyBytes)
+
+		configWatch = NewWatch(tlsConfigPath, certPath, keyPath)
+	})
+
+	It("should fail if config was not loaded", func() {
+		_, err := configWatch.GetConfig()
+		Expect(err).To(MatchError("tls profile not loaded"))
+	})
+
+	It("should load config from file", func() {
+		configWatch.Reload()
+
+		config, err := configWatch.GetConfig()
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(config.CipherSuites).To(ConsistOf(
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+		))
+		Expect(config.MinVersion).To(Equal(uint16(tls.VersionTLS12)))
+	})
+
+	It("should use default config if file does not exist", func() {
+		Expect(os.Remove(tlsConfigPath)).ToNot(HaveOccurred())
+		configWatch.Reload()
+
+		config, err := configWatch.GetConfig()
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(config.CipherSuites).To(BeEmpty())
+		Expect(config.MinVersion).To(BeZero())
+	})
+
+	It("should load certificate from file", func() {
+		configWatch.Reload()
+
+		config, err := configWatch.GetConfig()
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(config.Certificates).To(HaveLen(1))
+		Expect(config.Certificates[0].Leaf.DNSNames).To(ConsistOf(certHostName))
+	})
+
+	Context("watching", func() {
+		BeforeEach(func() {
+			configWatch.Reload()
+
+			done := make(chan struct{})
+			DeferCleanup(func() {
+				close(done)
+			})
+
+			go func() {
+				defer GinkgoRecover()
+				Expect(configWatch.Run(done)).To(Succeed())
+			}()
+
+			// Wait for a short time to let the watch goroutine run
+			runtime.Gosched()
+			Eventually(configWatch.IsRunning, time.Second, 50*time.Millisecond).Should(BeTrue())
+		})
+
+		It("should reload config on change", func() {
+			func() {
+				configFile, err := os.Create(tlsConfigPath)
+				Expect(err).ToNot(HaveOccurred())
+				defer configFile.Close()
+
+				tlsProfile := &v1alpha1.TlsSecurityProfile{
+					Type:   ocpconfigv1.TLSProfileModernType,
+					Modern: &ocpconfigv1.ModernTLSProfile{},
+				}
+				Expect(json.NewEncoder(configFile).Encode(tlsProfile)).To(Succeed())
+			}()
+
+			Eventually(func(g Gomega) {
```

Comment:
Make sure a different config was returned before the change?

Reply:
done